hydra/_internal/core_plugins/basic_launcher.py

```diff
@@ -36,23 +38,31 @@ def __init__(self) -> None:
         self.config: Optional[DictConfig] = None
         self.task_function: Optional[TaskFunction] = None
         self.hydra_context: Optional[HydraContext] = None
+        # BasicLauncher supports Hydra 1.0 style setup for compatibility with 3rd party Sweeper
+        self.config_loader: Optional[ConfigLoader] = None
```

Comment:
I don't quiet understand why you need to change the BasicLauncher here.
My understanding is that all instantiation and `setup()` call  are in the Plugins class and that changes there are sufficient to maintain compatibility with third party sweepers.

Reply:
You are right - I've removed the special conditions for BasicLauncher.